src/main/java/org/opensearch/knn/index/codec/util/CodecBuilder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.knn.index.codec.util;
+
+import lombok.AllArgsConstructor;
+import org.apache.lucene.codecs.Codec;
+import org.opensearch.index.mapper.MapperService;
+import org.opensearch.knn.index.codec.KNN910Codec.KNN910Codec;
+
+/**
+ * Abstracts builder logic for plugin codecs
+ */
+public abstract class CodecBuilder {
+    Codec userCodec;
+
+    /**
+     * Set user defined codec for plugin
+     * @param userCodec
+     * @return
+     */
+    public CodecBuilder userCodec(Codec userCodec) {
+        this.userCodec = userCodec;
+        return this;
+    }
+
+    /**
+     * Builds instance of codec, implementation is specific for each codec version
+     * @return
+     */
+    public abstract Codec build();
+
+    /**
+     * Implements builder abstraction for KNN91Codec
+     */
+    @AllArgsConstructor
+    public static class KNN91CodecBuilder extends CodecBuilder {
+        private final MapperService mapperService;
+
+        @Override
+        public Codec build() {
+            return new KNN910Codec(userCodec);
+        }
```

Comment:
not sure what is the use of the parameter MapperService here? Are we expecting when the dense_vector field we come we will use the mapper service to find out what exact codec to use?

Please improve the documentation with what we want achieve here.

Reply:
In this PR I'm making it available for codec and codec factory. In next PR I'm going to change the codec and extend knnVector format through the PerFieldMappingPostingFormatCodec mechanism, similar to what we did for core in dense_vector PR 
 https://github.com/opensearch-project/OpenSearch/pull/3659/files#r915126068. Let me update the description to reflect exact purpose of mapping service param

Reply:
updated PR description

Reply:
Just make sure that you are providing enough documentation and reason behind the changes in the Java doc.

Reply:
ack